app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerEffectHandler.kt

```diff
@@ -28,13 +36,30 @@ class TextInputDatePickerEffectHandler @AssistedInject constructor(
         .addAction(HideDateErrorMessage::class.java, uiActions::hideErrorMessage, schedulersProvider.ui())
         .addConsumer(ShowDateValidationError::class.java, ::applyValidationErrors, schedulersProvider.ui())
         .addConsumer(UserEnteredDateSelected::class.java, { uiActions.userEnteredDateSelected(it.userEnteredDate) }, schedulersProvider.ui())
+        .addTransformer(PrefilledDate::class.java, prefilledDate(schedulersProvider.ui()))
         .build()
   }
 
   private fun applyValidationErrors(effect: ShowDateValidationError) {
     showDateValidationErrors(effect.dateValidation)
   }
 
+  private fun prefilledDate(scheduler: Scheduler): ObservableTransformer<PrefilledDate, TextInputDatePickerEvent> {
+    return ObservableTransformer { prefilledDate ->
+      prefilledDate
+          .map(::getCurrentDateOrPrefilledDate)
+          .observeOn(scheduler)
+          .doOnNext(uiActions::setDateOnInputFields)
+          .map { DatePrefilled(it) }
+    }
+  }
+
+  private fun getCurrentDateOrPrefilledDate(prefillDate: PrefilledDate?): LocalDate {
+    val nowDate = Instant.now(userClock)
+    val nowLocalDate = LocalDateTime.ofInstant(nowDate, ZoneOffset.UTC).toLocalDate()
+    return prefillDate?.date ?: nowLocalDate
```

Comment:
It seems like irrespective of whether or not the prefill date is present we are creating the current date. Instead of elvis operator can we use if/else condition to make sure we are only creating current date object when prefill date is not present.
```kt
private fun getCurrentDateOrPrefilledDate(prefillDate: PrefilledDate): LocalDate {
  return if (prefillDate.date != null) {
    prefillDate.date
  } else {
    val instant = Instant.now(userClock)
    instant.toLocalDateAtZone(userClock.zone)
  }
}
```